backend/src/services/secret/secret-queue.ts

```diff
@@ -930,6 +930,11 @@ export const secretQueueFactory = ({
               }
             });
 
+            // re-throw error to re-run job unless final attempt, then log and send fail email
+            if (job.attemptsStarted !== job.opts.attempts) {
```

Comment:
@scott-ray-wilson Don't this throw still triggers the failed integration email.

Reply:
@akhilmhdh it will only send the email if it fails on the final attempt (ie when the error isn't re-thrown)

Reply:
That's beacuse there sendFailedIntegration sync has a debounce. But it still triggered on first try.

Even if your throw from catch, finally will execute in JS event cycle